### PR TITLE
Move pkg/k8sutil to internal/k8sutil

### PR DESCRIPTION
**Pull request by hasbro17**
**Description of the change:**
Move `pkg/k8sutil` to `internal/k8sutil` to remove public APIs for `v1.0.0` release

**Motivation for the change:**
Related to #3327

### CODE REVIEW

File: cmd/ansible-operator/main.go [hunk 1]
```diff
@@ -37,13 +37,13 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/manager/signals"
```
Comment (camilamacedo86) [inline cmd/ansible-operator/main.go:39]: MIssing the fragment it is a breaking change besides the I believe that the funcs used here by the users is to get the watchNamespace and the operator name which will no longer be useful in the new layout.

Comment (hasbro17) [reply cmd/ansible-operator/main.go:39]: Done

**Comment by joelanford** [comment]
Should we combine `internal/k8sutil` and `internal/util/k8sutil`?

**Comment by hasbro17** [comment]
@joelanford Good point, I would say yes. I'll move it to `internal/util/k8sutil`.

**Pull request merged**